src/celldega/sp/__init__.py

```diff
@@ -0,0 +1,359 @@
+    - n_rings: number of concentric rings to create
+    - band_width: desired band width in microns
+    - json_fname: output filename for GeoJSON
+    """
+    
+    # 1. Calculate conversion factor
+    micron_poly = gdf_micron[gdf_micron['roi'] == roi_name]['geometry'].iloc[0]
+    pixel_poly = gdf_pixel[gdf_pixel['roi'] == roi_name]['geometry'].iloc[0]
+    conversion_factor = (micron_poly.area / pixel_poly.area) ** 0.5
+
+    # 2. Create bands in pixel space (scaled by conversion factor)
+    gdf_pixel = gdf_pixel.loc[gdf_pixel["roi"] == roi_name].reset_index()
+    gdf_bands_pixel = create_concentric_rings(gdf_pixel, n_rings, band_width/conversion_factor)
+    
+    # 3. Save to GeoJSON
+    gdf_bands_pixel.to_file(json_fname, driver='GeoJSON')
+    return gdf_bands_pixel
+
+
+def assign_distinct_colors(spatial_region, color_metric='cat', alpha=100):
+    """
+    Assign perceptually distinct colors to features based on band values
+    Supports >12 distinguishable colors using HSV color space cycling
+    """
+    bands = [f['properties'][color_metric] for f in spatial_region['features']]
+    min_band, max_band = min(bands), max(bands)
+    
+    for feature in spatial_region['features']:
+        band = feature['properties'][color_metric]
+        
+        # Normalize band value (handle NaN/None)
+        try:
+            norm = (float(band) - min_band) / (max_band - min_band) if max_band > min_band else 0.5
+        except (TypeError, ValueError):
+            norm = 0.5  # Fallback for invalid values
+        
+        # Generate distinct color using HSV space
+        hue = norm * 0.9  # 0.9 avoids red-purple wrap which looks similar
+        saturation = 0.8 + (norm * 0.2)  # Vary saturation slightly
+        value = 0.7 + (norm * 0.3)  # Vary brightness
+        
+        # Convert to RGB
+        r, g, b = colorsys.hsv_to_rgb(hue, saturation, value)
+        
+        # Apply to feature (scale 0-255)
+        feature['properties']['color'] = [
+            int(r * 255),
+            int(g * 255),
+            int(b * 255),
+            alpha
+        ]
+
+def landscape_ist_with_spatial_region(gdf_pixel, path_landscape_files, spatial_region_json_fname, color_metric, tech='Xenium', alpha=50):
+    """
+    Create a landscape object with a spatial region and assign colors based on a metric.
+    Args:
+        gdf_pixel: GeoDataFrame containing pixel data.
+        path_landscape_files: Path to landscape files.
+        spatial_region_json_fname: Filename for the spatial region GeoJSON.
+        color_metric: Metric for color assignment.
+        tech: Technology used (default is 'Xenium').
+        alpha: Alpha value for color transparency (default is 50).
+    Returns:
+        landscape_ist: A Landscape object with the spatial region and assigned colors.
+    """
+
+    # save alphashape json
+    gdf_pixel[color_metric] = gdf_pixel[color_metric].astype(int)
+    gdf_pixel.to_file(spatial_region_json_fname, driver='GeoJSON')
```

Comment:
we do not need to save to a file and then load this file.

Reply:
this function is removed now